Fix for issue #275

bponsler opened the pull request:
Fix for issue #275 (State error calculation not accounting for wrapping joint positions).

Created an additional test case to test a robot with wrapping continuous joint angles.

bponsler commented:
This is for indigo since I don't have a quick way to compile/test for kinetic.

mathias-luedtke commented:
I have restarted [one job](https://travis-ci.org/ros-controls/ros_controllers/jobs/232655977) because of an unrelated error:
```
[diff_drive_controller.rosunit-diff_drive_odom_frame_test/testOdomTopic][FAILURE]
/root/catkin_ws/src/ros_controllers/diff_drive_controller/test/diff_drive_odom_frame_test.cpp:73
Value of: "new_odom"
Expected: odom_msg.header.frame_id.c_str()
Which is: ""
```

bmagyar commented on joint_trajectory_controller/test/joint_trajectory_controller_wrapping_test.cpp:
```diff
@@ -0,0 +1,279 @@
+///////////////////////////////////////////////////////////////////////////////
+// Copyright (C) 2013, PAL Robotics S.L.
```
You may want to adjust the name here

bponsler replied:
Who do you want this copyright attributed to? I'll also adjust the year to 2017 unless you disagree.

bmagyar replied:
Up to you. If it was on your own time, you can leave it like that or if it's within a company you may want to add them there. It's a test, hence it doesn't matter much.

bponsler replied:
I'm fine with it as is, thanks for the suggestion though!

graiola commented:
I try to run the fix on kinetic to see how it behaves...


... tests are fine on kinetic.

bmagyar commented:
Are you running this on Indigo? Do you have trouble on Indigo because of this? 
I'm a bit reluctant to add new stuff to Indigo that potentially changes behaviour but this can be considered bug fixing.

graiola commented:
Thanks for the pr @bponsler ! Could you please split the commit between the fix and the test? In this way we can cherry-pick the test on kinetic.

bponsler committed:
Fix for issue #275

Fix for issue #275 (State error calculation not accounting for wrapping joint positions)

bponsler committed:
Added tests for issue #275

bponsler commented:
Sure thing. Let me know if you want me to do anything else

graiola commented:
Thank you Brett!

bmagyar commented:
Thanks Brett!
